x-pack/plugins/security_solution/server/lib/detection_engine/notifications/rules_notification_alert_type.test.ts

```diff
@@ -79,6 +79,84 @@ describe('rules_notification_alert_type', () => {
       );
     });
 
+    it('should resolve results_link when meta is undefined to use "/app/security"', async () => {
```

Comment:
I always go back and forth with the wording - I find myself getting confused sometimes (might just be me). In tests when we say `x is undefined` should we be explicitly be checking that for example, `meta: undefined` or are we testing that the property itself is not there.

Reply:
I guess it's one in the same. 🤯

Reply:
It's technically not the same but we treat it a lot of the times as the same.

For example:

```ts
const a = { b: 1 }
Object.getOwnPropertyDescriptor(a, 'b')
{value: 1, writable: true, enumerable: true, configurable: true}

a.b = undefined;
Object.getOwnPropertyDescriptor(a, 'b')
{value: undefined, writable: true, enumerable: true, configurable: true}

delete a.b;
Object.getOwnPropertyDescriptor(a, 'b')
undefined
```

Technically you can see that the key is no longer there compared to if I just set it to undefined.

You can do this too:

```ts
a.b = undefined;
Object.keys(a)
["b"]
```

And really the key is still there. But for testing and stuff we might not be as strict when we could be. I sometimes try to use delete to be more exact or do a type of omit.